### PR TITLE
Fixed problems with double spaces in logs for `S3Common`

### PR DESCRIPTION
**Pull request by excitoon**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Improved logging of S3 errors, no more double spaces in case of empty keys and buckets.

### CODE REVIEW

**Comment by alesapin** [comment]
Why `backQuote`? Maybe normal `quoteString` would be better?

**Comment by alexey-milovidov** [comment]
We need to fix the unrelated issue found by Undefined Behaviour Sanitizer before merging this PR:

> ../src/Common/assert_cast.h:50:12:  runtime error: downcast of address 0x7f1bac307bb0 which does not point  to an object of type 'const DB::ColumnConst'

File: src/IO/S3Common.cpp [hunk 1]
```diff
@@ -631,7 +633,7 @@ namespace S3
             /// https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-s3-bucket-naming-requirements.html
             if (bucket.length() < 3 || bucket.length() > 63)
                 throw Exception(
```
Comment (alexey-milovidov) [inline src/IO/S3Common.cpp:635]: Let's also change to fmt-style Exception constructor.

Comment (excitoon) [reply src/IO/S3Common.cpp]: +

**Comment by alexey-milovidov** [comment]
> Stress test (thread) — Fatal message in clickhouse-server.log 

What is this message and why it is appeared?

**Comment by excitoon** [comment]
@alexey-milovidov main process was killed with SIGSEGV, probably because of sanitizer checks. However I can not see anything in `stderr.log`. I may guess that it was restarted externally and rewrote `stderr.log`.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**